src/java.base/share/classes/java/util/jar/JarFile.java

```diff
@@ -850,21 +853,33 @@ public synchronized InputStream getInputStream(ZipEntry ze)
             if (jv == null)
                 return super.getInputStream(ze);
         }
-
+        // Return null InputStream when the specified entry is not found in the
+        // Jar
+        var je = verifiableEntry(ze);
+        if (je == null) {
+            return null;
+        }
         // wrap a verifier stream around the real stream
         return new JarVerifier.VerifierStream(
-            getManifestFromReference(),
-            verifiableEntry(ze),
-            super.getInputStream(ze),
-            jv);
+                getManifestFromReference(),
+                je,
+                super.getInputStream(ze),
+                jv);
+
     }
 
-    private JarEntry verifiableEntry(ZipEntry ze) {
+    private JarEntry verifiableEntry(ZipEntry ze) throws ZipException {
         if (ze instanceof JarFileEntry) {
             // assure the name and entry match for verification
             return ((JarFileEntry)ze).realEntry();
         }
-        ze = getJarEntry(ze.getName());
+        // ZipEntry::getName should not return null
+        var entryName = ze.getName();
+        if (entryName != null) {
+            ze = getJarEntry(entryName);
+        } else {
+            throw new ZipException("ZipEntry::getName returned null");
```

Comment:
Throwing ZipException when ZipEntry::getName returns null is still surprising but not terrible.  The spec for getInputStream specifies ZipException for when a zip file format occurs so we might have to extend that to add "or the zip entry name is null".

Reply:
If you  would like me to update the ZipException to clarify this I can.   The original issue was due to a format error in the CEN so the current wording covers that.  It does not cover the case where ZipEntry is subclassed and ZipEntry::getName() returns null which is what your suggested wording would address.

Please note the above change addresses the signed jar scenario.  I can add an additional check in JarFile::getInputStream to check for null from ZipEntry::getName so that it covers unsigned jars and signed jars not being verified.

Please let me know your preference

Reply:
Per a discussion with Sean and Alan,  we are now returning null for the InputStream to be consistent with ZipFile::getInputStream and how unsigned jars are handled